src/boot/efi/xbootldr.c

```diff
@@ -68,7 +68,7 @@ static bool verify_gpt(union GptHeaderBuffer *gpt_header_buffer, EFI_LBA lba_exp
         if (h->MyLBA != lba_expected)
                 return false;
 
-        if (h->SizeOfPartitionEntry < sizeof(EFI_PARTITION_ENTRY))
+        if (h->SizeOfPartitionEntry % sizeof(EFI_PARTITION_ENTRY))
```

Comment:
as per coding style, please don't use C's downgrade-to-bool feature when not dealing with booleans, boolean-like stuff or pointers. For arithmetic values, please spell out what you are trying to say, i.e. write:

```c
if (h->SizeOfPartitionEntry % sizeof(EFI_PARTITION_ENTRY) != 0)
```